hack/validate-migration.sh

```diff
@@ -0,0 +1,332 @@
+resolve_migrations_parent_dir() {
+    local -r migration_file=$1
+    local -r dir_path=${migration_file%/*}  # remove file name
+    echo "${dir_path%/*}"  # remove path component migrations/
+}
+
+check_migrations_is_in_task_version_specific_dir() {
+    local -r migration_file=$1
+    local -r parent_dir=$(resolve_migrations_parent_dir "$migration_file")
+    local -r result=$(find task/*/* -type d -regex "$parent_dir" -print -quit)
+    if [ -z "$result" ]; then
+        echo "${FUNCNAME[0]}: migrations/ directory is not created in a task version-specific directory. Current is under $parent_dir. To fix it, move it to a path like task/task-1/0.1/." >&2
+        exit 1
+    fi
+}
+
+# Check that version within the migration file name must match the task version
+# in task label .metadata.labels."app.kubernetes.io/version".
+check_version_match() {
+    local -r migration_file=$1
+    local -r task_dir=$(resolve_migrations_parent_dir "$migration_file")
+
+    local task_name=${task_dir%/*}  # remove version part
+    task_name=${task_name##*/}  # remove all path components before the name
+
+    local task_version=
+    local -r label=".metadata.labels.\"${LABEL_TASK_VERSION}\""
+
+    if is_normal_task "$task_dir" "$task_name" ; then
+        task_version=$(yq "$label" "${task_dir}/${task_name}.yaml")
+    elif is_kustomized_task "$task_dir" "$task_name"; then
+        task_version=$(kubectl kustomize "$task_dir" | yq "$label")
+    else
+        exit 1
+    fi
+
+    if [ "${migration_file%/*}/${task_version}.sh" == "$migration_file" ]; then
+        return 0
+    fi
+
+    echo -n "${FUNCNAME[0]}: migration file does not match the task version '${task_version}'. " >&2
+    echo "Bump version in label '${LABEL_TASK_VERSION}' to match the migration." >&2
+
+    return 1
+}
+
+is_on_topic_branch() {
+    if [ "$(git branch --show-current)" != "$DEFAULT_BRANCH" ]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+is_migration_file() {
+    local -r file_path=$1
+    if [[ "$file_path" =~ /migrations/[0-9.]+\.sh$ ]]; then
+        return 0
+    fi
+    return 1
+}
+
+# Output migration file included in the branch
+# The file name is output to stdout with relative path to the root of the repository.
+# Generally, there should be one, but if multiple migration files are
+# discovered, it will be checked later. # No argument is required. Function
+# inspects changed files from current branch directly.
+list_migration_files() {
+    git diff --name-status "$(git merge-base HEAD "$DEFAULT_BRANCH")" | \
+        while read -r status origin_path; do
+            if ! is_migration_file "$origin_path"; then
+                continue
+            fi
+            case "$status" in
+                A)  # file is added
+                    echo "$origin_path"
+                    ;;
+                D | M)
+                    echo "It is not allowed to delete or modify existing migration file: $origin_path" >&2
```

Comment:
It would be nice to suggest what to do instead. I think the solution would to be bump the patch version and add a new migration that fixes the broken migration?